File: Python/String/split function/Maximum Number of Words You Can Type.py
class Solution(object):
    def canBeTypedWords(self, text, brokenLetters):
        """
        :type text: str
        :type brokenLetters: str
        :rtype: int
        """
        cnt=0
        broken=set(brokenLetters)
        for word in text.split():
            ok =True      # Assume each word is valid (ok = True).

            # Check every character of the word: If a character exists in the broken set → the word cannot be typed (ok = False) → stop checking further.
            for ch in word:
                if ch in broken:
                    ok=False
                    break
            if ok:    # If ok remains True, it means no broken letters were found → increment counter.
                cnt+=1
        return cnt
        

File: CPP/String/stringstream/Maximum Number of Words You Can Type.cpp
class Solution {
public:
    int canBeTypedWords(string text, string brokenLetters) {
        int cnt = 0;
        int sz = brokenLetters.size();
        unordered_set<char> broken(brokenLetters.begin(), brokenLetters.end());
        stringstream ss(text);
        string word;
        while (ss >> word){
            bool ok =true;
            for(auto ch:word){
                if(broken.count(ch)){
                    ok=false;
                    break;
                }
            }
            if(ok)cnt++;
        }
        return cnt;
    }
};
